examples/bootstrap_hwdefinition.yaml

```diff
@@ -1,36 +1,17 @@
 ---
-# BSD License
+# Copyright 2017 AT&T Intellectual Property.  All other rights reserved.
 #
-# Copyright (c) 2017, AT&T Intellectual Property.  All other rights reserved.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
 #
-# Redistribution and use in source and binary forms, with or
-# without modification, are permitted provided that the following
-# conditions are met:
+#     http://www.apache.org/licenses/LICENSE-2.0
 #
-# 1. Redistributions of source code must retain the above copyright
-# notice, this list of conditions and the following disclaimer.
-# 2. Redistributions in binary form must reproduce the above
-# copyright notice, this list of conditions and the following
-# disclaimer in the documentation and/or other materials provided
-# with the distribution.
-# 3. All advertising materials mentioning features or use of this
-# software must display the following acknowledgement:  This
-# product includes software developed by the AT&T.
-# 4. Neither the name of AT&T nor the names of its contributors
-# may be used to endorse or promote products derived from this
-# software without specific prior written permission.
-#
-# THIS SOFTWARE IS PROVIDED BY AT&T INTELLECTUAL PROPERTY ''AS IS'' AND
-# ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
-# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
-# PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL AT&T INTELLECTUAL PROPERTY
-# BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY,
-# OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
-# SUBSTITUTE GOODS OR SERVICES;  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
-# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
-# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
-# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
-# THE POSSIBILITY OF SUCH DAMAGE.
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
 #############################################################################
 #
 # bootstrap_hwdefinition.yaml - Definitions of server hardware layout
```

Comment:
one suggestion that i would like to make, is to take a play out of kubernetes book with flexible api/deployment models.

for example: define a version for API, and  type...like this

```
apiVersion: v1
kind: hardware
metadata:
  name: sample-metadata
  data: sample
spec:
 - name: HPGen8v3
   # Vendor of the server chassis
   vendor: HP
   # Generation of the chassis model
   generation: 8
   # Version of the chassis model within its generation - not version of the hardware definition
   version: 3
   # The certified version of the chassis BIOS
   bios_version: 2.2.3
...
...
```

This should allow us to be very flexible when we change/call API's (between versions) and allow for greater flexibility by expanding the `kind` field.